packages/orchestra/lib/client.ts

```diff
@@ -0,0 +1,84 @@
+import type { Json } from '@nangohq/scheduler';
+import { route as scheduleRoute } from './routes/v1/schedule.js';
+import { route as outputRoute } from './routes/v1/taskId/output.js';
+import type { Result } from '@nangohq/utils';
+import { Ok, Err, routeFetch } from '@nangohq/utils';
+
+interface SchedulingProps {
+    name: string;
+    groupKey: string;
+    retry: {
+        count: number;
+        max: number;
+    };
+    timeoutsInSecs: {
+        createdToStarted: number;
+        startedToCompleted: number;
+        heartbeat: number;
+    };
+    args: {
+        name: string;
+        connection: {
+            id: number;
+            provider_config_key: string;
+            environment_id: number;
+        };
+        activityLogId: number;
+        input: Json;
+    };
+}
+
+export class OrchestraClient {
+    private baseUrl: string;
+    private fetchTimeoutMs: number;
+
+    constructor({ baseUrl, fetchTimeoutMs = 120_000 }: { baseUrl: string; fetchTimeoutMs?: number }) {
+        this.baseUrl = baseUrl;
+        this.fetchTimeoutMs = fetchTimeoutMs;
+    }
+
+    public async schedule(props: SchedulingProps): Promise<Result<{ taskId: string }>> {
+        const res = await routeFetch(
+            this.baseUrl,
+            scheduleRoute
+        )({
+            body: {
+                scheduling: 'immediate',
+                name: props.name,
+                groupKey: props.groupKey,
+                retry: props.retry,
+                timeoutsInSecs: props.timeoutsInSecs,
+                args: props.args
+            }
+        });
+        if ('error' in res) {
+            return Err(`${res.error.code}: ${res.error.message || 'Unknown error'}`);
+        } else {
+            return Ok(res);
+        }
+    }
+
+    async execute(props: Pick<Partial<SchedulingProps>, 'name' | 'groupKey' | 'args'>): Promise<Result<Json>> {
+        const scheduleProps = {
+            retry: { count: 0, max: 0 },
+            timeoutsInSecs: { createdToStarted: 30, startedToCompleted: 30, heartbeat: 60 },
+            ...props
+        } as SchedulingProps;
+        const res = await this.schedule(scheduleProps);
+        if (res.isErr()) {
+            return res;
+        }
+        const taskId = res.value.taskId;
+        const start = Date.now();
+        const timeoutInMs = this.fetchTimeoutMs;
+        while (Date.now() - start < timeoutInMs) {
```

Comment:
naive implementation for now where output is requested in a loop until an error or output is receive. I will implement long polling before we make it globally available